CMakeLists.txt

```diff
@@ -641,7 +641,6 @@ if(IREE_BUILD_PYTHON_BINDINGS)
   # seems to be robust generally.
   # See: https://reviews.llvm.org/D118148
   # If building Python packages, we have a hard requirement on 3.9+.
-  find_package(Python3 3.9 COMPONENTS Interpreter Development NumPy)
```

Comment:
> We therefore skip searching for the latter as this might be the root for issue with the current macOS builds (see #19591).

I don't see how that could be related? One of the sample builds (https://github.com/iree-org/iree/actions/runs/12514308966/job/34910189766#step:8:77) already configures with explicit Python executables set (`'-DPython3_EXECUTABLE=/Users/runner/hostedtoolcache/Python/3.11.9/x64/bin/python3' '-DPYTHON_EXECUTABLE=/Users/runner/hostedtoolcache/Python/3.11.9/x64/bin/python3'`), so any number of `find_package` commands should be fine?

Reply:
That comes from a discussion on Discord but it turned out this is not related.